### PR TITLE
[Foundation] Add some missing C bindings in NSFileManager.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Add the bindings since Enviroment does not provide all/same information.

Fixes: https://github.com/xamarin/xamarin-macios/issues/3378

### CODE REVIEW

File: src/Foundation/NSFileManager.cs [hunk 1]
```diff
@@ -425,6 +426,59 @@ internal static NSFileSystemAttributes FromDictionary (NSDictionary dict)
 	}		
 	
 	public partial class NSFileManager {
+
+		[DllImport (Constants.FoundationLibrary)]
+		static extern IntPtr NSUserName ();
+
+		public static string UserName {
+			get {
+				using (var nsstring = ObjCRuntime.Runtime.GetNSObject<NSString> (NSUserName ()))
```
Comment (chamons) [inline src/Foundation/NSFileManager.cs]: How about the new style using format for these?

File: tests/monotouch-test/Foundation/FileManagerTest.cs [hunk 2]
```diff
@@ -37,6 +37,23 @@ public class NSFileManagerTest {
 			}
 		}
 
+		// we might believe that Envioment.UserName os the same as NSFileManager.UserName, but it is not. On the simulator for
+		// example, NSFileManager.UserName is an empty string while mono returns 'somebody'
+		[Test]
+		public void GetUserNameTest () => Assert.IsNotNull (NSFileManager.UserName);
```
Comment (chamons) [inline tests/monotouch-test/Foundation/FileManagerTest.cs]: Cute using arrows here, but I'd suggest moving to a block to make stepping a bit easier if we need to debug (?)

File: src/Foundation/NSFileManager.cs [hunk 3]
```diff
@@ -425,6 +426,59 @@ internal static NSFileSystemAttributes FromDictionary (NSDictionary dict)
...
+
+		[DllImport (Constants.FoundationLibrary)]
+		static extern IntPtr NSHomeDirectoryForUser (/* NSString */IntPtr userName);
+
+		public static string GetHomeDirectory (string userName) {
+			if (userName == null) 
+				throw new ArgumentNullException (nameof (userName));
+
+			using (var nsstring = new NSString (userName))
+			using (var homeDir = ObjCRuntime.Runtime.GetNSObject<NSString> (NSHomeDirectoryForUser (nsstring.Handle)))
```
Comment (prmallic) [inline src/Foundation/NSFileManager.cs]:
```suggestion
			using (var homeDir = ObjCRuntime.Runtime.GetNSObject<NSString> (NSHomeDirectoryForUser (nsstring.GetHandle ())))
```

File: src/Foundation/NSFileManager.cs [hunk 4]
```diff
@@ -425,6 +426,59 @@ internal static NSFileSystemAttributes FromDictionary (NSDictionary dict)
 	}		
 	
 	public partial class NSFileManager {
+
+		[DllImport (Constants.FoundationLibrary)]
+		static extern IntPtr NSUserName ();
+
+		public static string UserName {
```
Comment (prmallic) [inline src/Foundation/NSFileManager.cs]: A bit confused, or maybe i'm looking at the wrong place. Checking the header of `NSFileManager.h` - doesn't have `NSUserName`. Could only find `FOUNDATION_EXPORT NSString *NSUserName(void);` in `NSPathUtilities.h`. How do we know its a static property of the class?

File: src/Foundation/NSFileManager.cs (hunk 1 above)
Comment (mandel-macaque) [reply src/Foundation/NSFileManager.cs]: Will work the same way. using will be until the end of the get. Problem with me is that I don't trust that new using syntactic sugar. I don't trust myself with the scope ;)

File: src/Foundation/NSFileManager.cs (hunk 4 above)
Comment (mandel-macaque) [reply src/Foundation/NSFileManager.cs]: It is not a static property in the class. Is a C function. But I need to place it somewhere and NSFileManager is a class that makes sense. 

The C function does not need the object, so we don't care. But we need and object to expose the function.

File: tests/monotouch-test/Foundation/FileManagerTest.cs (hunk 2 above)
Comment (mandel-macaque) [reply tests/monotouch-test/Foundation/FileManagerTest.cs]: In the test? No need

File: src/Foundation/NSFileManager.cs (hunk 1 above)
Comment (Therzok) [reply src/Foundation/NSFileManager.cs]: Maybe use [NSString.FromHandle](https://github.com/xamarin/xamarin-macios/blob/fffa889f22fad3c220e1b50a5884e077b0556e99/src/Foundation/NSString.cs#L196
):
```csharp
				return NSString.FromHandle (NSUserName ());
```

**Comment by mandel-macaque** [comment]
Failures are known: https://github.com/xamarin/maccore/issues/581

**Pull request merged**